### ISSUE TITLE
Backup fails on bulk loader data.

### ISSUE THREAD

**Issue by martinmr** [dgraph-io/dgraph#3831]
Starting with commit 30ad3a4f44da9a0681d7f6f526886ba9e1f75604, backup hangs forever when trying to backup an instance on which the bulk loader was used (even if mutations are added later). @danielmai reported that the rollup function seems to be taking a long time.
This commit introduces the backwards-compatible format, which must rollup the lists. Before that, rollup was not called during backup.

- What version of Dgraph are you using?
master

- Have you tried reproducing the issue with latest release?
yes

- Steps to reproduce the issue (command/config used to run Dgraph).
1. Load 1 million dataset into dgraph using the bulk loader.
1. Send a backup request (either to minio/s3 or the local filesystem).

- Expected behaviour and actual result.
Backup completes

**Comment by martinmr** [comment]
I debugged a little bit more and I've narrowed down the issue a bit. To summarize:

- Rollup is not the issue, as it completes successfully and the output appears to be reasonable.
- The function `toBackupList` in `ee/backup/backup.go` never exits. This function takes an iterator and goes through the posting lists, rolling them up and converting them to the new format. However, the iteration loops forever. I've tried to add more itr.Next in the places where the loop returns or breaks but that has not worked.
- I still have no idea why this only happens when bulk loader is involved. Manual and automated tests using the client all work fine.
